File: Strings/ReverseString.java
//Reverse the given string in java

public String reverseString(String s) {
    char[] charArray = s.toCharArray();
    int left = 0, right = s.length() - 1;
    while (left < right) {
        char temp = charArray[left];
        charArray[left] = charArray[right];
        charArray[right] = temp;
        left++;
        right--;
    }
    return new String(charArray);
}

File: Recursion/reverseString.js
var reverseString = function(s) {
  //Recursive approach
    if(s=="")
    return"";
    else
    return reverseString(s.substr(1)) + s.charAt(0);

  //2 pointer approach
    // let left = 0, right = s.length- 1;
    // while (left < right) {
    //     let temp = s[left];
    //     s[left] = s[right];
    //     s[right] = temp;
    //     left++;
    //     right--;
    // }
    // return s;
};

